File: SkaiLab.Invoice.Models/PurchaseOrder.cs
using Microsoft.AspNetCore.SignalR;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SkaiLab.Invoice.Models
{
    public class PurchaseOrder: Expense
    {
      
    }
    
    public class PurchaseOrderItem:ExpenseItem
    {
        public long? ProductId { get; set; }
        public Product Product { get; set; }
        public long? LocationId { get; set; }
    }
    public class PurchaseOrderForUpdate
    {
        public PurchaseOrder PurchaseOrder { get; set; }
        public List<Currency> Currencies { get; set; }
        public List<Location> Locations { get; set; }
        public List<Tax> Taxes { get; set; }
        public int BaseCurrencyId { get; set; }
        public Currency TaxCurrency { get; set; }

    }
}

File: SkaiLab.Invoice/ClientApp/src/app/models/purchaseOrder.ts
import { Expense } from "./expense";
import { Product } from "./product";
import { Tax } from "./tax";

export class PurchaseOrder extends Expense{
    
    constructor(){
        super();
       
    }
}

export class PurchaseOrderItem{
    id:number;
    productId:number;
    quantity:number;
    taxId?:number;
    lineTotal:number;
    lineTotalIncludeTax:number;
    product:Product;
    tax:Tax;
    discountRate?:number;
    unitPrice:number;
    locationId:number;
    description:string;
    constructor(){
        this.id=0;
        this.productId=null;
        this.product=new Product();
        this.quantity=null;
        this.taxId=null;
        this.discountRate=null;
        this.tax=null;
        this.lineTotal=0;
        this.lineTotalIncludeTax=0;
        this.unitPrice=0;
        this.locationId=null;
        this.description="";
    }

}